pandas/tests/io/test_sql.py

```diff
@@ -1906,1102 +1924,1356 @@ def test_api_read_sql_duplicate_columns(conn, request):
-                chunksize=3,
-            )
-            expected = self.dtype_backend_expected(string_storage, dtype_backend)
-            for result in iterator:
-                tm.assert_frame_equal(result, expected)
-
-    @pytest.mark.parametrize("func", ["read_sql", "read_sql_table"])
-    def test_read_sql_dtype_backend_table(self, string_storage, func, dtype_backend):
-        # GH#50048
-        table = "test"
-        df = self.dtype_backend_data()
-        df.to_sql(name=table, con=self.conn, index=False, if_exists="replace")
-
-        with pd.option_context("mode.string_storage", string_storage):
-            result = getattr(pd, func)(table, self.conn, dtype_backend=dtype_backend)
-        expected = self.dtype_backend_expected(string_storage, dtype_backend)
-        tm.assert_frame_equal(result, expected)
 
-        with pd.option_context("mode.string_storage", string_storage):
-            iterator = getattr(pd, func)(
-                table,
-                self.conn,
-                dtype_backend=dtype_backend,
-                chunksize=3,
-            )
-            expected = self.dtype_backend_expected(string_storage, dtype_backend)
-            for result in iterator:
-                tm.assert_frame_equal(result, expected)
-
-    @pytest.mark.parametrize("func", ["read_sql", "read_sql_table", "read_sql_query"])
-    def test_read_sql_invalid_dtype_backend_table(self, func):
-        table = "test"
-        df = self.dtype_backend_data()
-        df.to_sql(name=table, con=self.conn, index=False, if_exists="replace")
-
-        msg = (
-            "dtype_backend numpy is invalid, only 'numpy_nullable' and "
-            "'pyarrow' are allowed."
-        )
-        with pytest.raises(ValueError, match=msg):
-            getattr(pd, func)(table, self.conn, dtype_backend="numpy")
+@pytest.fixture
+def dtype_backend_data() -> DataFrame:
+    return DataFrame(
+        {
+            "a": Series([1, np.nan, 3], dtype="Int64"),
+            "b": Series([1, 2, 3], dtype="Int64"),
+            "c": Series([1.5, np.nan, 2.5], dtype="Float64"),
+            "d": Series([1.5, 2.0, 2.5], dtype="Float64"),
+            "e": [True, False, None],
+            "f": [True, False, True],
+            "g": ["a", "b", "c"],
+            "h": ["a", "b", None],
+        }
+    )
 
-    def dtype_backend_data(self) -> DataFrame:
-        return DataFrame(
-            {
-                "a": Series([1, np.nan, 3], dtype="Int64"),
-                "b": Series([1, 2, 3], dtype="Int64"),
-                "c": Series([1.5, np.nan, 2.5], dtype="Float64"),
-                "d": Series([1.5, 2.0, 2.5], dtype="Float64"),
-                "e": [True, False, None],
-                "f": [True, False, True],
-                "g": ["a", "b", "c"],
-                "h": ["a", "b", None],
-            }
-        )
 
-    def dtype_backend_expected(self, storage, dtype_backend) -> DataFrame:
+@pytest.fixture
+def dtype_backend_expected():
```

Comment:
This is the pattern I would ultimately like to follow with the types / iris tables, i.e. just create a separate fixture that takes the connection as an argument and can build / drop the tables from there  @mroeschke